files/archive.py

```diff
@@ -0,0 +1,385 @@
+            remove = dict(required=False, default=False, type='bool'),
+        ),
+        add_file_common_args=True,
+        supports_check_mode=True,
+    )
+
+    params = module.params
+    paths = params['path']
+    dest = params['dest']
+    remove = params['remove']
+    expanded_paths = []
+    compression = params['compression']
+    globby = False
+    changed = False
+    state = 'absent'
+
+    # Simple or archive file compression (inapplicable with 'zip' since it's always an archive)
+    archive = False
+    successes = []
+
+    for i, path in enumerate(paths):
+        path = os.path.expanduser(path)
+
+        # Detect glob-like characters
+        if any((c in set('*?')) for c in path):
+            expanded_paths = expanded_paths + glob.glob(path)
+        else:
+            expanded_paths.append(path)
+
+    if len(expanded_paths) == 0:
+        return module.fail_json(path=', '.join(paths), expanded_paths=', '.join(expanded_paths), msg='Error, no source paths were found')
+
+    # If we actually matched multiple files or TRIED to, then
+    # treat this as a multi-file archive
+    archive = globby or os.path.isdir(expanded_paths[0]) or len(expanded_paths) > 1
+
+    # Default created file name (for single-file archives) to
+    # <file>.<compression>
+    if dest:
+        dest = os.path.expanduser(dest)
+    elif not archive:
+        dest = '%s.%s' % (expanded_paths[0], compression)
+
+    # Force archives to specify 'dest'
+    if archive and not dest:
+        module.fail_json(dest=dest, path=', '.join(paths), msg='Error, must specify "dest" when archiving multiple files or trees')
+
+    archive_paths = []
+    missing = []
+    exclude = []
+    arcroot = ''
+
+    for path in expanded_paths:
+        # Use the longest common directory name among all the files
+        # as the archive root path
+        if arcroot == '':
+            arcroot = os.path.dirname(path) + os.sep
+        else:
+            for i in xrange(len(arcroot)):
+                if path[i] != arcroot[i]:
+                    break
+
+            if i < len(arcroot):
+                arcroot = os.path.dirname(arcroot[0:i+1])
+
+            arcroot += os.sep
+
+        # Don't allow archives to be created anywhere within paths to be removed
+        if remove and os.path.isdir(path) and dest.startswith(path):
+            module.fail_json(path=', '.join(paths), msg='Error, created archive can not be contained in source paths when remove=True')
+
+        if os.path.lexists(path):
+            archive_paths.append(path)
+        else:
+            missing.append(path)
+
+    # No source files were found but the named archive exists: are we 'compress' or 'archive' now?
+    if len(missing) == len(expanded_paths) and dest and os.path.exists(dest):
+        # Just check the filename to know if it's an archive or simple compressed file
+        if re.search(r'(\.tar\.gz|\.tgz|.tbz2|\.tar\.bz2|\.zip)$', os.path.basename(dest), re.IGNORECASE):
```

Comment:
I'd add an 'import re'

The 're' module gets imported into this namespace via the module_utils.basic * include, but would prefer the
explicit import.